connectors/sources/network_drive.py

```diff
@@ -232,6 +233,26 @@ def fetch_members(self, group_name):
         return self.parse_output(members)
 
 
+class SMBSession(metaclass=Singleton):
```

Comment:
Why make it singleton?

IMO it becomes problematic potentially if you have several network drive connectors in same deployment. Can you make sure that syncing 2 network drives at the same time works when doing these changes?

Reply:
Truly sorry, I was trying multiple approaches and somehow missed to remove the singleton approach.